utils/make_endpoint.cpp

```diff
@@ -11,11 +11,43 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
-#include "make_endpoint.hpp"
 
+#include <include/bluzelle.hpp>
+#include <utils/make_endpoint.hpp>
 
-boost::asio::ip::tcp::endpoint
+
+std::optional<boost::asio::ip::tcp::endpoint>
```

Comment:
This is where the magic happens, I think. Where we make sure the endpoint is valid.